File: frontend/src/services/timetable/timetableSettings.ts
import api from "../../lib/api";

export interface TimetableSettings {
  academic_year_id: string;
  school_start_time: string;
  school_end_time: string;
  number_of_periods: number;
  period_duration_minutes: number;
  interval_duration_minutes: number;
}

export const timetableSettingsApi = {
  getByYear: (academicYearId: string) =>
    api.get<TimetableSettings>(`/timetable-settings/${academicYearId}`).then((r) => r.data),

  upsert: (data: TimetableSettings) =>
    api.put<TimetableSettings>("/timetable-settings", data).then((r) => r.data),
};

File: backend/internal/models/timetable/timetable_settings.go
package timetable

import "github.com/google/uuid"

type UpsertTimetableSettingsRequest struct {
	AcademicYearID          uuid.UUID `json:"academic_year_id" binding:"required"`
	SchoolStartTime         string    `json:"school_start_time" binding:"required"`
	SchoolEndTime           string    `json:"school_end_time" binding:"required"`
	NumberOfPeriods         int32     `json:"number_of_periods" binding:"required"`
	PeriodDurationMinutes   int32     `json:"period_duration_minutes" binding:"required"`
	IntervalDurationMinutes int32     `json:"interval_duration_minutes" binding:"required"`
}

type TimetableSettingsResponse struct {
	AcademicYearID          uuid.UUID `json:"academic_year_id"`
	SchoolStartTime         string    `json:"school_start_time"`
	SchoolEndTime           string    `json:"school_end_time"`
	NumberOfPeriods         int32     `json:"number_of_periods"`
	PeriodDurationMinutes   int32     `json:"period_duration_minutes"`
	IntervalDurationMinutes int32     `json:"interval_duration_minutes"`
}
